Source/WebCore/platform/SharedBuffer.cpp

```diff
@@ -385,17 +385,17 @@ Vector<uint8_t> FragmentedSharedBuffer::read(size_t offset, size_t length) const
     return data;
 }
 
-void FragmentedSharedBuffer::copyTo(void* destination, size_t length) const
+void FragmentedSharedBuffer::copyTo(std::span<uint8_t> destination) const
 {
-    return copyTo(destination, 0, length);
+    return copyTo(destination, 0);
 }
 
-void FragmentedSharedBuffer::copyTo(void* destination, size_t offset, size_t length) const
+void FragmentedSharedBuffer::copyTo(std::span<uint8_t> destination, size_t offset) const
 {
-    ASSERT(length + offset <= size());
+    ASSERT(offset + destination.size() <= size());
```

Comment:
This assertion doesn’t seem quite right. There’s nothing that prevents an absurdly large value of `offset` from causing an overflow. Luckily, the non-assertion code doesn’t have that problem. Ideally we should either remove this assertion or rewrite it or break it into multiple assertions so it can’t be fooled by overflow. I’m not sure we need the assertion since the code handles the edge cases correctly.